onadata/apps/logger/signals.py

```diff
@@ -1,9 +1,44 @@
 # coding: utf-8
 import logging
-from django.db.models.signals import pre_delete
+
+from django.db.models import F
+from django.db.models.signals import (
+    pre_delete,
+    post_save,
+)
 from django.dispatch import receiver
 
 from onadata.apps.logger.models.attachment import Attachment
+from onadata.apps.logger.models.xform import XForm
+from onadata.apps.main.models.user_profile import UserProfile
+
+
+@receiver(pre_delete, sender=Attachment)
+def delete_user_profile_attachment_storage_bytes(instance, **kwargs):
+    """
+    Update the attachment_storage_bytes field in the UserProfile model
+    when an attachment is deleted
+    """
+    file_size = instance.media_file.size
+    owner_profile = instance.instance.xform.user.profile.pk
+    queryset = UserProfile.objects.filter(pk=owner_profile)
+    queryset.update(
+        attachment_storage_bytes=F('attachment_storage_bytes') - file_size
+    )
+
+
+@receiver(pre_delete, sender=Attachment)
+def delete_xform_attachment_storage_bytes(instance, **kwargs):
```

Comment:
this is nitpicky, but the method name sounds like an xform is being deleted. maybe something like `delete_attachment_subtract_xform_storage_bytes` would be appropriate? a similar naming convention could be applied to the previous handler, `delete_user_profile_attachment_storage_bytes`.